packages/dependency-extraction-webpack-plugin/index.js

```diff
@@ -130,37 +137,61 @@ class DependencyExtractionWebpackPlugin {
 
 				const runtimeChunk = entrypoint.getRuntimeChunk();
 
-				// Get a stable, stringified representation of the WordPress script asset.
-				const assetString = this.stringify( {
+				const assetData = {
 					dependencies: Array.from(
 						entrypointExternalizedWpDeps
 					).sort(),
 					version: runtimeChunk.hash,
-				} );
+				};
+
+				// Get a stable, stringified representation of the WordPress script asset.
+				const assetString = this.stringify( assetData );
 
 				// Determine a filename for the asset file.
 				const [ filename, query ] = entrypointName.split( '?', 2 );
-				const assetFilename = compilation
-					.getPath( outputFilename, {
-						chunk: runtimeChunk,
-						filename,
-						query,
-						basename: basename( filename ),
-						contentHash: createHash( 'md4' )
-							.update( assetString )
-							.digest( 'hex' ),
-					} )
-					.replace(
-						/\.js$/i,
-						'.asset.' + ( outputFormat === 'php' ? 'php' : 'json' )
-					);
+				const buildFilename = compilation.getPath( outputFilename, {
+					chunk: runtimeChunk,
+					filename,
+					query,
+					basename: basename( filename ),
+					contentHash: createHash( 'md4' )
+						.update( assetString )
+						.digest( 'hex' ),
+				} );
+
+				if ( combineAssets ) {
+					combinedAssetsData[ buildFilename ] = assetData;
```

Comment:
This is producing results like `fileA.js`, but I wonder what the most desirable format is 🤔

```php
array('fileA.js' => array( /* … */ ) );
```

What do you think about using `entrypointName` directly?

```php
array('fileA' => array( /* … */ ) );
```

Reply:
`fileA` - I don't think if WordPress core will be able to do anything with the name of the entry point. We still would need to generate the path. WordPress uses kebab case for script handles.

Reply:
Maybe what you've already done here makes the most sense. We'd expect this to match the entrypoint ouput file?

I'd expect plugin authors to use this by finding the entrypoint file to enqueue, then looking that file up in the assets array. Accessing by filename may be the best way 👍 

I don't really know what will be best without playing with this or seeing it used in the wild. Disregard my previous comment 🙂